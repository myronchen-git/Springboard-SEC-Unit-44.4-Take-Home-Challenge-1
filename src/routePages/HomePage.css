.HomePage {
  text-align: center;
}

.HomePage .btn {
  margin: 0.5em;
}

.HomePage .btn > a {
  color: inherit;
  text-decoration: inherit;
}
