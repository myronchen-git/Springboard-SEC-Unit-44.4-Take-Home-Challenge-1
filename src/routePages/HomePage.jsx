import { Link } from 'react-router-dom';
import { Button } from 'reactstrap';

import './HomePage.css';

// ==================================================

function HomePage() {
  return (
    <main className="HomePage">
      <h1>Yodlr Design Challenge</h1>
      <Button color="primary" outline>
        <Link to="/signup">Registration Page</Link>
      </Button>
      <Button color="primary" outline>
        <Link to="/admin">Admin Page</Link>
      </Button>
    </main>
  );
}

// ==================================================

export default HomePage;
