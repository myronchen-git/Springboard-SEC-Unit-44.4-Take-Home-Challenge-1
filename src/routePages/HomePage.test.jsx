import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';

import HomePage from './HomePage';

// ==================================================

describe('HomePage', () => {
  it('renders.', () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
  });

  it('matches snapshot.', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains correct links.', () => {
    // Act
    const { getByText } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    // Assert
    expect(getByText('Registration Page')).toBeVisible();
    expect(getByText('Admin Page')).toBeVisible();
  });
});
