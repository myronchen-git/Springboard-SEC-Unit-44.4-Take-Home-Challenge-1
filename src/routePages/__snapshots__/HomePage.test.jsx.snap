// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > matches snapshot. 1`] = `
<DocumentFragment>
  <main
    class="HomePage"
  >
    <h1>
      Yodlr Design Challenge
    </h1>
    <button
      class="btn btn-outline-primary"
    >
      <a
        data-discover="true"
        href="/signup"
      >
        Registration Page
      </a>
    </button>
    <button
      class="btn btn-outline-primary"
    >
      <a
        data-discover="true"
        href="/admin"
      >
        Admin Page
      </a>
    </button>
  </main>
</DocumentFragment>
`;
